### DISCUSSION TITLE
Found a typo?

### DISCUSSION THREAD

**Discussion by jamiepine** [spacedriveapp/spacedrive#142] [General]
Any changes to spelling, grammar or the wording structure of the README, code comments or documentation can be suggested here to avoid pull request clutter. Thank you!

**Comment by yadunandanbhat** [comment]
The "links will be added once release available" is wrong grammar, it should be "Links will be added once the release is available". And the note about the app being in development is added twice, which is redundant. Can i open a pr for this?

**Reply by maxichrome** [reply]
It's definitely not _explicitly incorrect_ and could be considered proper in [headlinese](https://en.wikipedia.org/wiki/Headline#Headlinese), which in this case could be considered proper for brevity's sake.

Where is the redundant note about the app in development? Can you link the lines? That might warrant an issue's creation

**Reply by yadunandanbhat** [reply]
![image](https://user-images.githubusercontent.com/54680617/167995766-13d56070-5280-4e09-b0d9-a3697a96dd3b.png)
See the above image

**Reply by maxichrome** [reply]
I'll open an issue to discuss if/how we want to condense these

**Reply by yadunandanbhat** [reply]
And in the monorepo structure, some links point to their respective websites, while some point to the readme itself, which is kinda odd.
See below image.
![image](https://user-images.githubusercontent.com/54680617/167996046-9d4bb655-4192-4148-91ab-c063dd133728.png)
Every link in the app sub heading point to their respective website, but everything else just points to "https://github.com/spacedriveapp/spacedrive#"

**Reply by jamiepine** [reply]
> I'll open an issue to discuss if/how we want to condense these

not necessary, fixed already

**Reply by maxichrome** [reply]
> > I'll open an issue to discuss if/how we want to condense these
> 
> not necessary, fixed already

How do you mean "fixed already"?

**Reply by maxichrome** [reply]
> And in the monorepo structure, some links point to their respective websites, while some point to the readme itself, which is kinda odd.
> See below image.
> ![image](https://user-images.githubusercontent.com/54680617/167996046-9d4bb655-4192-4148-91ab-c063dd133728.png)
> Every link in the app sub heading point to their respective website, but everything else just points to "https://github.com/spacedriveapp/spacedrive#"

The second notice you brought up earlier actually addresses this.

**Reply by jamiepine** [reply]
https://github.com/spacedriveapp/spacedrive/commit/ce058ab49cb96eaa189ebdddd616595579d82d46 
https://github.com/spacedriveapp/spacedrive/commit/34ebbf4364e58bd97f956d3544b95f8ff8c538cb

**Reply by maxichrome** [reply]
> [ce058ab](https://github.com/spacedriveapp/spacedrive/commit/ce058ab49cb96eaa189ebdddd616595579d82d46)
> [34ebbf4](https://github.com/spacedriveapp/spacedrive/commit/34ebbf4364e58bd97f956d3544b95f8ff8c538cb)

That second commit removes the notice about links being broken

**Reply by jamiepine** [reply]
they should be fixed asap

**Reply by yadunandanbhat** [reply]
cool!

**Comment by leozaffre** [comment]
The link to the `contributing guide` under **Developer Guide** is broken.
https://github.com/spacedriveapp

**Comment by ghost** [comment]
Feild should be field here: https://github.com/spacedriveapp/spacedrive/blob/main/core/prisma/schema.prisma#L183

**Comment by s4mbre** [comment]
README.md
Line 88
keybind should be key bind


docs/architecture/virtual-filesystem.md
Line 41
prioritised should be prioritized 


docs/product/roadmap.md
Line 18
keybind should be key bind

**Reply by jamiepine** [reply]
I know "key bind" is probably grammatically correct, but I think it looks awkward and I prefer using it as a single word.

**Comment by hulxv** [comment]
I suggest making a GitHub workflow by using [typos](https://github.com/crate-ci/typos) to check if there are typos or not. I think that's better.

**Reply by oscartbeaumont** [reply]
We are gonna do this at some point. Currently, we have a lefthook script for it but it doesn't seem to be catching typos very well.

**Comment by schneedotdev** [comment]
Line 62
"VSFS" should be "VDFS" at "VSFS is not restricted to a single machine

<img width="1119" alt="Screen Shot 2022-07-14 at 2 11 55 PM" src="https://user-images.githubusercontent.com/77141303/179053145-dd87e44f-30d8-448f-a051-96e818ff6403.png">

**Comment by z1cheng** [comment]
Typos:
1. core/Cargo.toml, [Line 11](https://github.com/spacedriveapp/spacedrive/blob/main/core/Cargo.toml#L11): controlls -> controls
2. core/src/job/jobs.rs, [Line 46](https://github.com/spacedriveapp/spacedrive/blob/main/core/src/job/jobs.rs#L46) and [Line 55](https://github.com/spacedriveapp/spacedrive/blob/main/core/src/job/jobs.rs#L55): internal_reciever -> internal_receiver
3. core/src/util/db.rs, [Line 18](https://github.com/spacedriveapp/spacedrive/blob/main/core/src/util/db.rs#L18): occured -> occurred

You can also check this [PR](https://github.com/spacedriveapp/spacedrive/pull/339/files)

**Reply by Brendonovich** [reply]
[5e30331](https://github.com/spacedriveapp/spacedrive/commit/5e30331dad93f657f3129f9e7bf58369d25692fc)

**Comment by christophermorin** [comment]
Hello, 
A few typos in some of the docs:

1. README, Line 114 - 'completly' -> completely
2. docs/architecture/search, Line 5 - 'simotanious' -> simultaneous
3. docs/architecture/spaces, Line 3b - 'spacially' -> spatially
4. docs/architecture/sync, Line 5 - 'Logicial' -> Logical

**Reply by Brendonovich** [reply]
[5e30331](https://github.com/spacedriveapp/spacedrive/commit/5e30331dad93f657f3129f9e7bf58369d25692fc)

**Comment by JJJJJJJJJJJJJJJl** [comment]
@introduction.md - Line 14

Fixing typo where "optimized" is miswritten as "optimizes".

Also, defining the word decentralized as a system in which information is shared (and trusted) across multiple network users, I don't think describing Spacedrive as a decentralized design is the most accurate move. But, this is probably minor.

**Reply by jamiepine** [reply]
I agree. I diverted from referring to it as "distributed" because I felt that if the nodes synchronize without a master then it must be "decentralized", but that isn't entirely true. Our data sync design works similarly to consensus, all the nodes will be eventually consistent without a server or master node to resolve conflicts — however I understand that decentralized typically refers to blockchain apps, which this certainly not. Thanks for bringing this up, appreciate it.

**Comment by simonmurdock** [comment]
https://www.spacedrive.com/docs/product/getting-started/introduction

"Create photo albums that you'll never **loose**, effortlessly"

**Reply by Brendonovich** [reply]
'loose' means not being contained, rigid or confined. 'lose' is the right word for that sentence.

**Comment by shoebob15** [comment]
In [the node docs](https://github.com/spacedriveapp/spacedrive/edit/main/docs/developers/architecture/libraries.md), the hyperlink that is supposed to go to the nodes doc instead links back to the current page. In the source docs, it doesn't link to anything: `A library is the database that Spacedrive stores all file structures and metadata. It can be synchronized with other [Nodes]().`. This also happens on the website.

**Reply by utkubakir** [reply]
docs are wip right now, once we have some content, we will go over all links. Thanks!

**Comment by gBasil** [comment]
"its" should be "it's" in "and best of all, its open source and entirely free." in the file `docs/product/getting-started/introduction.md`.

**Comment by tomheaton** [comment]
README.md - Line `107`:
```diff
- - `config`: `eslint` configurations (includes `eslint-config-next`, `eslint-config-prettier` and all `tsconfig.json` configs used throughout the monorepo.
+ - `config`: `eslint` configurations (includes `eslint-config-next`, `eslint-config-prettier` and all `tsconfig.json` configs used throughout the monorepo).
```

**Reply by utkubakir** [reply]
fixed, thank you

**Comment by Crilum** [comment]
Not sure where it is in the code, but this is definitely not correct grammar:
![image](https://github.com/spacedriveapp/spacedrive/assets/91354257/e683af01-b65a-4fa1-8880-d8bd532979fa)

I think it might be supposed to say `Scanned 2 more files or directories;`?

**Reply by utkubakir** [reply]
fixed! thank you

**Comment by alexsmacd** [comment]
The Changelog link in the footer of SpaceDrive.com is broken. It should go to https://www.spacedrive.com/docs/changelog/alpha/0.1.4, but it links to https://www.spacedrive.com/docs/changelog/beta/0.1.0.

Question/Suggestion: Is there a way to setup a redirect so that https://www.spacedrive.com/docs/changelog/ goes to the latest version?

**Reply by utkubakir** [reply]
Yeah we should be to redirect users to latest changelog. Thanks for the suggestion!

**Comment by ParamBirje** [comment]
The "App Crashed" screen has the string "Check the console (CMD/CTRL + OPTION + i) for stack trace." which is generic for all operating systems.

Specifically on Win/Linux, the shortcut to open the console is assigned to CTRL+SHIFT+i, we need to add SHIFT in the message on the crash screen to make it consistent with the instructions in CONTRIBUTING.md

Thus making the new string "Check the console (CMD/CTRL + OPTION/SHIFT + i) for stack trace."

Can I put up a PR for it for this? [The line in ErrorFallback.tsx](https://github.com/spacedriveapp/spacedrive/blob/7078bc05597ec2a6445cc13c8aa97f0773b05ee1/interface/ErrorFallback.tsx#L107C59-L107C59)

**Reply by utkubakir** [reply]
fixed, thank you!

**Comment by Raghav-45** [comment]
https://www.spacedrive.com/docs/product/guides/interface

> Let's start by getting a clear understanding of the Spacedrive interface. Designed to mimic what you know and love with existing file explorers, but with powerful features and a universal _`desgin`_ across operating systems.

Design spelling is wrong

**Reply by utkubakir** [reply]
fixed on #2317 thank you!

**Comment by David-Kyrat** [comment]
## What ?

I detected a typo in actual code (`./scripts/setup.sh`)  i don't know if i should report it here.

## When?

- When using fedora , i.e. dnf gets detected when running `./scripts/setup.sh`

## Where?

- On line 188 in

```bash
      # Tauri dependencies
      set -- openssl webkit2gtk4.1-devel openssl-dev curl wget file libappindicator-gtk3-devel librsvg2-devel libxdo-devel dbus-devel
```

`openssl-dev` should be `openssl-devel` (because dev packages are called "devel" on rpm based systems.)

like this

```bash
      set -- openssl webkit2gtk4.1-devel openssl-devel curl wget file libappindicator-gtk3-devel librsvg2-devel libxdo-devel dbus-devel
```

## How am I sure it's indeed a typo

- When i first ran the script, dnf gave me an error message like "package openssl-dev" does not exist.
- After doing the replacement on line 188, i ran the setup script again and it appear to have worked ( I got the message "Your machine has been setup for Spacedrive development!" at the end)

**Reply by iLynxcat** [reply]
@HeavenVolkoff @matheus-consoli can one of you confirm & possibly take a few minutes to resolve this?

**Comment by randompersonperson** [comment]
In [https://github.com/spacedriveapp/spacedrive/blob/main/docs/developers/architecture/vdfs.mdx](https://github.com/spacedriveapp/spacedrive/blob/main/docs/developers/architecture/vdfs.mdx), the last word should be "accommodate", not " accomedate".

**Reply by valeriewong052** [reply]
Fixed, thanks!

**Comment by randompersonperson** [comment]
In [https://github.com/spacedriveapp/spacedrive/blob/main/docs/developers/architecture/libraries.mdx](https://github.com/spacedriveapp/spacedrive/blob/main/docs/developers/architecture/libraries.mdx), 
it should be "You must create a library to use Spacedrive. Users can also choose to create more libraries, such as ones specifically for work. Spacedrive will run all your libraries in unison."

instead of "You must create a library to use Spacedrive. Users can also choose to create more librariesm, such as ones specifically for work. Spacedrive will run all your libraries in unison."

**Reply by valeriewong052** [reply]
Fixed!

**Comment by iLynxcat** [comment]
Locking and closing this discussion — in a PR I'll be submitting shortly tonight we'll take typo reports via GitHub Issues
